src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -112,46 +112,38 @@ void uct_rdmacm_iface_client_start_next_ep(uct_rdmacm_iface_t *iface)
 static void uct_rdmacm_client_handle_failure(uct_rdmacm_iface_t *iface, ucs_status_t status)
 {
     ucs_assert(!iface->is_server);
-    uct_rdmacm_ep_set_failed(&iface->super.super, &iface->ep->super.super, status);
+    if (iface->ep != UCT_RDMACM_IFACE_BLOCKED_NO_EP) {
+        uct_rdmacm_ep_set_failed(&iface->super.super, &iface->ep->super.super, status);
+    }
     uct_rdmacm_iface_client_start_next_ep(iface);
 }
 
 static void uct_rdmacm_iface_process_conn_req(uct_rdmacm_iface_t *iface,
                                               struct rdma_cm_event *event,
                                               struct sockaddr *remote_addr)
 {
-    uct_rdmacm_priv_data_hdr_t *hdr, cb_hdr;
-    ssize_t server_data_len;
+    uct_rdmacm_priv_data_hdr_t *hdr;
     struct rdma_conn_param conn_param;
     char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    ucs_status_t status;
 
     hdr = (uct_rdmacm_priv_data_hdr_t*) event->param.ud.private_data;
 
-    memset(&conn_param, 0, sizeof(conn_param));
-    conn_param.private_data = ucs_alloca(UCT_RDMACM_MAX_CONN_PRIV + sizeof(uct_rdmacm_priv_data_hdr_t));
 
     /* TODO check the iface's cb_flags to determine when to invoke this callback.
      * currently only UCT_CB_FLAG_ASYNC is supported so the cb is invoked from here */
-    server_data_len = iface->conn_request_cb(iface->conn_request_arg,
-                                             event->param.ud.private_data +
-                                             sizeof(uct_rdmacm_priv_data_hdr_t), /* private data */
-                                             hdr->length,                        /* length */
-                                             (void*)(conn_param.private_data +
-                                             sizeof(uct_rdmacm_priv_data_hdr_t)));
-    if (server_data_len < 0) {
-        ucs_error("rdmacm server (iface=%p cm_id=%p fd=%d) failed to fill private data",
-                   iface, event->id, iface->event_ch->fd);
+    status = iface->conn_request_cb(iface->conn_request_arg,
+                                    event->param.ud.private_data +
+                                    /* private data */
+                                    sizeof(uct_rdmacm_priv_data_hdr_t),
+                                    /* length */
+                                    hdr->length);
+    if (status != UCS_OK) {
         rdma_reject(event->id, NULL, 0);
         return;
     }
 
-    cb_hdr.length = server_data_len;
-    /* The private_data starts with the header of the user's private data and then
-     * the private data itself */
-    UCS_STATIC_ASSERT(sizeof(cb_hdr) == sizeof(uct_rdmacm_priv_data_hdr_t));
-    memcpy((void*)conn_param.private_data, &cb_hdr, sizeof(uct_rdmacm_priv_data_hdr_t));
-    conn_param.private_data_len = sizeof(uct_rdmacm_priv_data_hdr_t) + cb_hdr.length;
-
+    memset(&conn_param, 0, sizeof(conn_param));
```

Comment:
would be nice to set explicitly `conn_param.private_data_len = 0;`

Reply:
won't memset do that?

Reply:
it will, so maybe just add comment that we are not sending any data in the reply

Reply:
ok